nexus/types/src/deployment/planning_input.rs

```diff
@@ -474,16 +493,30 @@ pub enum SledFilter {
     /// Sleds that are eligible for discretionary services.
     Discretionary,
```

Comment:
Nit: `EligibleForDiscretionaryServices`?

Though actually, I might rename this to `EligibleForOptionalServices`? Idk. Just seems like being more explicit is useful with additional variants.

Reply:
I remember having concerns about "optional" because at the rack scale the services aren't optional.

Reply:
I like `EligibleForDiscretionaryServices`